app/gui/src/dashboard/layouts/Editor.tsx

```diff
@@ -1,67 +1,33 @@
 /** @file The container that launches the IDE. */
-import * as React from 'react'
-
-import * as reactQuery from '@tanstack/react-query'
-
-import * as appUtils from '#/appUtils'
-
+import * as errorBoundary from '#/components/ErrorBoundary'
+import * as suspense from '#/components/Suspense'
+import { useEventCallback } from '#/hooks/eventCallbackHooks'
 import * as gtagHooks from '#/hooks/gtagHooks'
 import * as projectHooks from '#/hooks/projectHooks'
-
 import * as backendProvider from '#/providers/BackendProvider'
 import type { LaunchedProject } from '#/providers/ProjectsProvider'
 import * as textProvider from '#/providers/TextProvider'
-
-import * as errorBoundary from '#/components/ErrorBoundary'
-import * as suspense from '#/components/Suspense'
-
-import type Backend from '#/services/Backend'
 import * as backendModule from '#/services/Backend'
-
-import { useEventCallback } from '#/hooks/eventCallbackHooks'
 import * as twMerge from '#/utilities/tailwindMerge'
+import * as reactQuery from '@tanstack/react-query'
+import * as React from 'react'
 import { useTimeoutCallback } from '../hooks/timeoutHooks'
-
-// ====================
-// === StringConfig ===
-// ====================
-
-/** A configuration in which values may be strings or nested configurations. */
-interface StringConfig {
-  readonly [key: string]: StringConfig | string
-}
-
-// ========================
-// === GraphEditorProps ===
-// ========================
+// eslint-disable-next-line no-restricted-syntax
+import ProjectViewTabVue from '@/ProjectViewTab.vue'
+import { applyPureVueInReact } from 'veaury'
+import type { AllowedComponentProps, VNodeProps } from 'vue'
+import type { ComponentProps } from 'vue-component-type-helpers'
 
 /** Props for the GUI editor root component. */
-export interface GraphEditorProps {
-  readonly config: StringConfig | null
-  readonly projectId: string
-  readonly hidden: boolean
-  readonly ignoreParamsRegex?: RegExp
-  readonly logEvent: (message: string, projectId?: string | null, metadata?: object | null) => void
-  readonly renameProject: (newName: string) => void
-  readonly projectBackend: Backend | null
-  readonly remoteBackend: Backend | null
-}
-
-// =========================
-// === GraphEditorRunner ===
-// =========================
+export type ProjectViewTabProps = Omit<
+  ComponentProps<typeof ProjectViewTabVue>,
+  keyof AllowedComponentProps | keyof VNodeProps
+>
 
-/**
- * The value passed from the entrypoint to the dashboard, which enables the dashboard to
- * open a new IDE instance.
- */
-export type GraphEditorRunner = React.ComponentType<GraphEditorProps>
-
-// =================
-// === Constants ===
-// =================
-
-const IGNORE_PARAMS_REGEX = new RegExp(`^${appUtils.SEARCH_PARAMS_PREFIX}(.+)$`)
+// eslint-disable-next-line no-restricted-syntax
+const ProjectViewTab = applyPureVueInReact(ProjectViewTabVue) as (
+  props: ProjectViewTabProps,
+) => JSX.Element
```

Comment:
Why is this cast necessary?

Reply:
Otherwise I get a typecheck error.
```
src/dashboard/layouts/Editor.tsx:212:11 - error TS2786: 'ProjectViewTab' cannot be used as a JSX component.
  Its type 'Function' is not a valid JSX element type.

212   return <ProjectViewTab key={key} {...appProps} />
              ~~~~~~~~~~~~~~
```
JSX element cannot be just any `Function`, and this is what is returned from veaury. If you know better way to fix this error, please let me know.

Reply:
That's annoying, but I guess fine for now. It looks like a typing bug in veaury.